### PR TITLE
Add Code changes to match the design of the List Recommendations API

### PR DESCRIPTION
**Pull request by bharathappali**

### CODE REVIEW

File: src/main/java/com/autotune/utils/AnalyzerConstants.java [hunk 1]
```diff
@@ -134,6 +134,33 @@ public enum ResourceSetting {
...
 
+    public enum RecommendationCategory {
+        duration_based,
+        profile_based
+    }
+
+    public enum RecommendationNotificationTypes {
+        INFO ("info", 1),
+        WARN ("warning", 2),
+        ERROR ("error", 3);
```
Comment (dinogun) [inline src/main/java/com/autotune/utils/AnalyzerConstants.java:145]: We have a `CRITICAL` that is missing here

**Comment by dinogun** [comment]
Please add the API.md changes as well

**Comment by bharathappali** [comment]
@dinogun Here is the current response from `listExperiments`

```
[
{
        "kubernetes_objects": [
            {
                "type": "deployment",
                "name": "tfb-qrh-deployment",
                "namespace": "default",
                "containers": [
                    {
                        "container_image_name": "kruize/tfb-qrh:1.13.2.F_et17",
                        "container_name": "tfb-server-1",
                        "recommendations": {
                            "2022-01-23T18:25:43.511Z": {
                                "duration_based": {
                                    "short_term": {
                                        "monitoring_start_time": "2022-01-22T18:25:43.511Z",
                                        "monitoring_end_time": "2022-01-23T18:25:43.511Z",
                                        "duration_in_hours": 0.0,
                                        "pods_count": 0,
                                        "confidence_level": 0.0,
                                        "config": {
                                            "limits": {
                                                "cpu": {
                                                    "amount": 6.51,
                                                    "format": "cores"
                                                },
                                                "memory": {
                                                    "amount": 982.5997506234414,
                                                    "format": "MiB"
                                                }
                                            },
                                            "requests": {
                                                "cpu": {
                                                    "amount": 1.03,
                                                    "format": "cores"
                                                },
                                                "memory": {
                                                    "amount": 123.6,
                                                    "format": "MiB"
                                                }
                                            }
                                        },
                                        "notifications": []
                                    },
                                    "medium_term": {
                                        "pods_count": 0,
                                        "confidence_level": 0.0,
                                        "notifications": [
                                            {
                                                "type": "info",
                                                "message": "There is not enough data available to generate a recommendation."
                                            }
                                        ]
                                    },
                                    "long_term": {
                                        "pods_count": 0,
                                        "confidence_level": 0.0,
                                        "notifications": [
                                            {
                                                "type": "info",
                                                "message": "There is not enough data available to generate a recommendation."
                                            }
                                        ]
                                    }
                                }
                            }
                        }
                    },
                    {
                        "container_image_name": "kruize/tfb-db:1.15",
                        "container_name": "tfb-server-0",
                        "recommendations": {
                            "2022-01-23T18:25:43.511Z": {
                                "duration_based": {
                                    "short_term": {
                                        "monitoring_start_time": "2022-01-22T18:25:43.511Z",
                                        "monitoring_end_time": "2022-01-23T18:25:43.511Z",
                                        "duration_in_hours": 0.0,
                                        "pods_count": 0,
                                        "confidence_level": 0.0,
                                        "config": {
                                            "limits": {
                                                "cpu": {
                                                    "amount": 6.51,
                                                    "format": "cores"
                                                },
                                                "memory": {
                                                    "amount": 982.5997506234414,
                                                    "format": "MiB"
                                                }
                                            },
                                            "requests": {
                                                "cpu": {
                                                    "amount": 1.03,
                                                    "format": "cores"
                                                },
                                                "memory": {
                                                    "amount": 123.6,
                                                    "format": "MiB"
                                                }
                                            }
                                        },
                                        "notifications": []
                                    },
                                    "medium_term": {
                                        "pods_count": 0,
                                        "confidence_level": 0.0,
                                        "notifications": [
                                            {
                                                "type": "info",
                                                "message": "There is not enough data available to generate a recommendation."
                                            }
                                        ]
                                    },
                                    "long_term": {
                                        "pods_count": 0,
                                        "confidence_level": 0.0,
                                        "notifications": [
                                            {
                                                "type": "info",
                                                "message": "There is not enough data available to generate a recommendation."
                                            }
                                        ]
                                    }
                                }
                            }
                        }
                    }
                ]
            }
        ],
        "version": "1.0",
        "experiment_name": "quarkus-resteasy-autotune-min-http-response-time-db"
    }
]
```

Please lemme know if you see any inconsistencies. My only observation was that if there are no notifications it's showing empty, I think it's ok for now and in clean up phase we can remove the section if it's empty.

File: src/main/java/com/autotune/utils/AnalyzerConstants.java (hunk 1 above)
Comment (bharathappali) [reply src/main/java/com/autotune/utils/AnalyzerConstants.java:154]: Sure will be adding that

Comment (bharathappali) [reply src/main/java/com/autotune/utils/AnalyzerConstants.java]: Thanks for the review @dinogun I have added `CRITICAL` type as well

**Comment by bharathappali** [comment]
> Please add the API.md changes as well

Added them in the latest commit

**Comment by bharathappali** [comment]
Ah! sorry got confused it's `listRecommendations`

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**